File: python/get_user_publication_following.py
#-- title: Get User's Publication Following
#-- description: Code to get publications that the given user is following.
#-- tags: python, medium_api, medium_api_py


# Import libraries
import os
from medium_api import Medium

#%%
# Get RAPIDAPI_KEY from the environment
api_key = os.getenv('RAPIDAPI_KEY')

#%%
# Create a `Medium` Object
medium = Medium(api_key)

#%%
# Create a "User" Object and fetch publication following
user = medium.user(username="nishu-jain")

user.fetch_publication_following()

#%%
# Iterate over user's publication following and print each publication's name
for publication in user.publication_following:
    print(publication.name)

File: javascript/getUserPublicationFollowing.js
// getUserPublicationFollowing.js - Fetches the publications a user is following

import Medium from 'medium-api-js';
import dotenv from 'dotenv';

dotenv.config();

const medium = new Medium(process.env.RAPIDAPI_KEY);

const userId = '1985b61817c3'; 

// Fetch the publications that the user is following based on their user ID
medium.getUserPublicationFollowing(userId)
    .then(data => console.log('User Publication Following:', data));
